python/plugins/processing/algs/otb/OtbAlgorithm.py

```diff
@@ -207,31 +211,41 @@ def processAlgorithm(self, parameters, context, feedback):
             param = self.parameterDefinition(k)
             if param.isDestination():
                 continue
-
-            if isinstance(param, (QgsProcessingParameterRasterLayer, QgsProcessingParameterVectorLayer)):
-                if isinstance(v, QgsMapLayer):
-                    value = '"{}"'.format(v.source())
-                else:
-                    value = '"{}"'.format(v)
-            elif isinstance(param, QgsProcessingParameterMultipleLayers):
-                value = ''
-                for item in v:
-                    value += '"{}" '.format(item)
-            elif isinstance(param, QgsProcessingParameterCrs):
-                crs = self.parameterAsCrs(parameters, param.name(), context)
-                value = crs.authid().split('EPSG:')[1]
-
-            elif isinstance(param, QgsProcessingParameterEnum):
+            if isinstance(param, QgsProcessingParameterEnum):
                 value = self.parameterAsEnum(parameters, param.name(), context)
+            elif isinstance(param, QgsProcessingParameterBoolean):
+                value = self.parameterAsBool(parameters, param.name(), context)
+            elif isinstance(param, QgsProcessingParameterCrs):
+                crsValue = self.parameterAsCrs(parameters, param.name(), context)
+                value = crsValue.authid().split('EPSG:')[1]
+            elif isinstance(param, QgsProcessingParameterFile):
+                value = self.parameterAsFile(parameters, param.name(), context)
+            elif isinstance(param, QgsProcessingParameterMultipleLayers):
+                layers = self.parameterAsLayerList(parameters, param.name(), context)
+                if layers is None or len(layers) == 0:
+                    continue
+                value = ' '.join(['"{}"'.format(layer.source()) for layer in layers])
```

Comment:
What about postgis/sql server/memory layers? You'll either need to handle these, e.g. some equivalent of https://github.com/qgis/QGIS/blob/master/python/plugins/processing/algs/gdal/GdalAlgorithm.py#L75, or raise an exception and inform users that only disk based layers are supported.